### ISSUE TITLE
Android Crash within EncryptWithEntity

### ISSUE THREAD

**Issue by yagottahavehart** [oblador/react-native-keychain#27]
Device: sm_n900v100
OS : 4.3

Javascript Call:
```
const Keychain = require('react-native-keychain');
Keychain.setInternetCredentials(serverString, userString, passString)
...
```

Stack:
> 
```
Fatal Exception: java.lang.RuntimeException: Could not invoke RNKeychainManager.setInternetCredentialsForServer
       at com.facebook.react.bridge.BaseJavaModule$JavaMethod.invoke(BaseJavaModule.java:281)
       at com.facebook.react.bridge.NativeModuleRegistry$ModuleDefinition.call(NativeModuleRegistry.java:185)
       at com.facebook.react.bridge.NativeModuleRegistry.call(NativeModuleRegistry.java:64)
       at com.facebook.react.bridge.CatalystInstanceImpl$NativeModulesReactCallback.call(CatalystInstanceImpl.java:420)
       at com.facebook.react.bridge.queue.NativeRunnableDeprecated.run(NativeRunnableDeprecated.java)
       at android.os.Handler.handleCallback(Handler.java:730)
       at android.os.Handler.dispatchMessage(Handler.java:92)
       at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:31)
       at android.os.Looper.loop(Looper.java:137)
       at com.facebook.react.bridge.queue.MessageQueueThreadImpl$3.run(MessageQueueThreadImpl.java:187)
       at java.lang.Thread.run(Thread.java:841)
Caused by java.lang.reflect.InvocationTargetException
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:525)
       at com.facebook.react.bridge.BaseJavaModule$JavaMethod.invoke(BaseJavaModule.java:268)
       at com.facebook.react.bridge.NativeModuleRegistry$ModuleDefinition.call(NativeModuleRegistry.java:185)
       at com.facebook.react.bridge.NativeModuleRegistry.call(NativeModuleRegistry.java:64)
       at com.facebook.react.bridge.CatalystInstanceImpl$NativeModulesReactCallback.call(CatalystInstanceImpl.java:420)
       at com.facebook.react.bridge.queue.NativeRunnableDeprecated.run(NativeRunnableDeprecated.java)
       at android.os.Handler.handleCallback(Handler.java:730)
       at android.os.Handler.dispatchMessage(Handler.java:92)
       at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:31)
       at android.os.Looper.loop(Looper.java:137)
       at com.facebook.react.bridge.queue.MessageQueueThreadImpl$3.run(MessageQueueThreadImpl.java:187)
       at java.lang.Thread.run(Thread.java:841)
Caused by java.lang.NoClassDefFoundError: java.nio.charset.StandardCharsets
       at com.oblador.keychain.KeychainModule.encryptWithEntity(KeychainModule.java:78)
       at com.oblador.keychain.KeychainModule.setGenericPasswordForService(KeychainModule.java:61)
       at com.oblador.keychain.KeychainModule.setInternetCredentialsForServer(KeychainModule.java:137)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:525)
       at com.facebook.react.bridge.BaseJavaModule$JavaMethod.invoke(BaseJavaModule.java:268)
       at com.facebook.react.bridge.NativeModuleRegistry$ModuleDefinition.call(NativeModuleRegistry.java:185)
       at com.facebook.react.bridge.NativeModuleRegistry.call(NativeModuleRegistry.java:64)
       at com.facebook.react.bridge.CatalystInstanceImpl$NativeModulesReactCallback.call(CatalystInstanceImpl.java:420)
       at com.facebook.react.bridge.queue.NativeRunnableDeprecated.run(NativeRunnableDeprecated.java)
       at android.os.Handler.handleCallback(Handler.java:730)
       at android.os.Handler.dispatchMessage(Handler.java:92)
       at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:31)
       at android.os.Looper.loop(Looper.java:137)
       at com.facebook.react.bridge.queue.MessageQueueThreadImpl$3.run(MessageQueueThreadImpl.java:187)
       at java.lang.Thread.run(Thread.java:841)```

**Comment by vonovak** [comment]
Looking at the stack trace I see Caused by java.lang.NoClassDefFoundError:
java.nio.charset.StandardCharsets. What java version are you using?
I'm not an android expert but it might be this is not available on your
Android version in which case a fix won't be difficult.

**Comment by yagottahavehart** [comment]
import java.nio.charset.StandardCharsets; is only in 19+

**Comment by vonovak** [comment]
@yagottahavehart could you ship over a PR?

**Comment by yagottahavehart** [comment]
@vonovak Done!

**Comment by oblador** [comment]
Fixed in [0.3.2](https://github.com/oblador/react-native-keychain/releases/tag/v0.3.2).
